File: HW1/Collection.java
import java.util.*;

public class Collection {

	ArrayList<Shape> shapes;
	int numShapes;
	boolean sortX, sortY, sortZ;

	public Collection () {
		this.numShapes = 0;
		this.shapes = new ArrayList<Shape>();
	}

	public void add (Shape item) {
		sortX = sortY = sortZ = false;
		numShapes++;
		shapes.add(item);
	}

	public void sort(boolean nsortX, boolean nsortY, boolean nsortZ) {
		sortX = nsortX;
		sortY = nsortY;
		sortZ = nsortZ;
		if (sortX) {
			Collections.sort(shapes, new XComparator()); 
		} else if (sortY) {
			Collections.sort(shapes, new YComparator()); 
		} else if (sortZ) {
			Collections.sort(shapes, new ZComparator()); 			
		}
	}

	private void displaySortOrder() {
		if (sortX)
			System.out.println ("Sorted by X co-ordinate"); 
		else if (sortY)
			System.out.println ("Sorted by Y co-ordinate"); 
		else if (sortY)
			System.out.println ("Sorted by Z co-ordinate"); 
		else
			System.out.println ("Shapes not sorted"); 
	}

	public void display() {
		System.out.println ("Type   of Shape : " + shapes.get(0).getName()); 
		System.out.println ("Number of Shape : " + numShapes); 
		displaySortOrder();
		for(Shape str: shapes){
			str.display();
	   }
		System.out.println ("------------------------------------------------------"); 
	}
}

File: HW2/Collection.py
from Shape import *
from Circle import *
from Square import *
from Triangle import *

class Collection:

	def __init__ (self):
		self._numShapes = 0;
		self._shapes = []
		self._sortX = False
		self._sortY = False
		self._sortZ = False

	def add (self, item):
		self._numShapes = self._numShapes+1
		self._shapes.append(item)

	def sort (self, nsortX, nsortY, nsortZ):
		self._sortX = nsortX;
		self._sortY = nsortY;
		self._sortZ = nsortZ;

		if (self._sortX): 
			self._shapes = sorted(self._shapes, key=lambda shape: shape.getX())
		if (self._sortY): 
			self._shapes = sorted(self._shapes, key=lambda shape: shape.getY())
		if (self._sortZ): 
			self._shapes = sorted(self._shapes, key=lambda shape: shape.getZ())

	def displaySortOrder(self):
		if (self._sortX):	
			print ("Sorted by X co-ordinate")
		if (self._sortY):
			print ("Sorted by Y co-ordinate") 
		if (self._sortZ):
			print ("Sorted by Z co-ordinate") 			

	def display(self):
		print ("--------------------------------------------------------------------")
		print ("Type of Shape : " + self._shapes[0].getName())
		self.displaySortOrder()
		for i in self._shapes:
			i.display()

	def displayArea(self):
		for i in self._shapes:
			print ("area : " + str(i.getArea()))